be/src/exec/pipeline/exchange/exchange_sink_operator.cpp

```diff
@@ -455,38 +487,46 @@ Status ExchangeSinkOperator::push_chunk(RuntimeState* state, const vectorized::C
                 }
             }
 
-            // Compute row indexes for each channel
-            _channel_row_idx_start_points.assign(num_channels + 1, 0);
-            for (uint16_t i = 0; i < num_rows; ++i) {
-                uint16_t channel_index = _hash_values[i] % num_channels;
-                _channel_row_idx_start_points[channel_index]++;
-                _hash_values[i] = channel_index;
+            // Compute row indexes for each channel's each shuffle
+            _channel_row_idx_start_points.assign(num_channels * _num_shuffles + 1, 0);
+
+            for (size_t i = 0; i < num_rows; ++i) {
+                auto channel_id = _hash_values[i] % num_channels;
+                auto driver_sequence = _hash_values[i] % _num_shuffles;
```

Comment:
driver_sequence is very confusing,  suggest to use `sub_channel_id` or `pipeline_channel_id`.

Reply:
`driver_sequence` is meaningful, it comes from `OperatorFactory::create`

```cpp
  class OperatorFactory {
  public:
      virtual OperatorPtr create(int32_t degree_of_parallelism, int32_t driver_sequence) = 0;
```